artifact/metadata.go

```diff
@@ -126,6 +311,42 @@ func (ti *TypeInfo) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
 
+type TypeInfoDepends struct {
+	RootfsChecksum string `json:"rootfs_image_checksum"`
+}
+
+type TypeInfoProvides struct {
+	RootfsChecksum string `json:"rootfs_image_checksum"`
+}
+
+// TypeInfoV3 provides information about the type of update contained within the
+// headerstructure.
+type TypeInfoV3 struct {
+	// Rootfs/Delta (Required).
+	Type string `json:"type"`
+	// Checksum of the image that needs to be installed on the device in order to
+	// apply the current update.
+	ArtifactDepends []TypeInfoDepends `json:"artifact_depends,omitempty"`
+	// Checksum of the image currently installed on the device.
+	ArtifactProvides []TypeInfoProvides `json:"artifact_provides,omitempty"`
```

Comment:
Am I correct that these two should be changed now after https://github.com/mendersoftware/mender-artifact/pull/139#discussion_r225507535 ?

Reply:
Well, if you mean that they should be changed to key-value pairs, I was a bit ahead of time (?), and thus this is the way it is already implemented.

Reply:
Well, if you mean that they should be changed to key-value pairs, I was a bit ahead of time (?), and thus this is the way it is already implemented.

Reply:
Well, if you mean that they should be changed to key-value pairs, I was a bit ahead of time (?), and thus this is the way it is already implemented.

Reply:
Well, if you mean that they should be changed to key-value pairs, I was a bit ahead of time (?), and thus this is the way it is already implemented.

Reply:
yes